alica_solver_interface/include/alica_solver_interface/Interval.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <algorithm>
+#include <ostream>
+namespace alica
+{
+template <typename T>
+class Interval
+{
+  public:
+    Interval() {}
+    Interval(T min, T max)
+        : _min(min)
+        , _max(max)
+    {
+    }
+
+    Interval<T>(const Interval<T>& o) = default;
+    Interval<T>& operator=(const Interval<T>& o) = default;
+
+    T getMin() const { return _min; }
+    T getMidPoint() const { return (_max + _min) / 2; }
+    T getMax() const { return _max; }
+    void limitTo(T min, T max)
+    {
+        _min = std::max<T>(min, _min);
+        _max = std::min<T>(max, _max);
+    }
+    void limitTo(const Interval<T> o)
+    {
+        _min = std::max<T>(o._min, _min);
+        _max = std::min<T>(o._max, _max);
+    }
+
+    Interval intersect(const Interval<T> o) const { return Interval(std::max<T>(_min, o._min), std::min<T>(_max, o._max)); }
+    bool contains(const Interval<T> o) const { return _min <= o._min && _max >= o._max; }
+    T clamp(T val) const { return std::max<T>(_min, std::min<T>(val, _max)); }
```

Comment:
this function returns min if the interval is invalid (min>max). What are the consequences of this? Should this function return an error instead?

Reply:
No it should not do any additional error checking. The result of clamping with an invalid interval is not defined, it could do an assert though.